### PR TITLE
feat: add client id to getserviceconfig call

### LINKED ISSUE #164: [Node SDK] Add client_id to GetServiceConfig call
Please see https://github.com/StanzaSystems/apis/pull/20

Here's the go impl for reference: https://github.com/StanzaSystems/sdk-go/pull/89

### PR DESCRIPTION
**Pull request by sre-is-carlos**
fixes #164

### CODE REVIEW

File: packages/node/src/hub/rest/createRestHubService.ts [hunk 1]
```diff
@@ -28,15 +28,16 @@ export const createRestHubService = ({ serviceName, serviceRelease, environment,
     prefix: '[REST Hub Service]'
   }, {
     getServiceMetadata: () => ({ serviceName, serviceRelease, environment, clientId }),
-    fetchServiceConfig: async ({ lastVersionSeen } = {}) => {
+    fetchServiceConfig: async (options) => {
```
Comment (sre-is-carlos) [inline packages/node/src/hub/rest/createRestHubService.ts:31]: Here, I changed the pattern to equal the 'grpc.'

**Review by jamie-kuppens-sre-is — APPROVED** [review]

**Comment by jamie-kuppens-sre-is** [comment]
I think this looks good but I'm curious what Chris thingks. @ktrz-sre-is Does this look good to you?

File: packages/node/src/hub/rest/createRestHubService.ts (hunk 1 above)
Comment (ktrz-sre-is) [reply packages/node/src/hub/rest/createRestHubService.ts:31]: 👍

**Review by ktrz-sre-is — CHANGES_REQUESTED** [review]
This looks good. But I think we need to modify `packages/node/src/service/startPollingConfigService.ts` to pass the `clientId` param down. Otherwise the client id is never going to be passed to the Hub Service

And maybe add some additional tests in `packages/node/src/service/fetchServiceConfig.test.ts`

File: packages/node/src/hub/api/serviceConfigResponse.ts [hunk 2]
```diff
@@ -41,12 +41,14 @@ const zSentinelConfig = z.object({
...
 
 const serviceConfigNoData = z.object({
+  clientId: z.string().optional(),
   version: z.string(),
   configDataSent: z.literal(false),
   config: z.null().optional()
 })
 
 const serviceConfigWithData = z.object({
+  clientId: z.string().optional(),
```
Comment (ktrz-sre-is) [inline packages/node/src/hub/api/serviceConfigResponse.ts:51]: This ZOD validation is modeling the response from the hub.  We just send a client id as a request param and it does not contain the client id in the response

File: packages/node/src/hub/model.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { type StanzaTokenResponse } from './api/stanzaTokenResponse'
 
 type DataSent<T extends { configDataSent: boolean }> = T extends { configDataSent: true } ? T : never
 
-export type ServiceConfig = Pick<DataSent<ServiceConfigResponse>, 'version' | 'config'>
+export type ServiceConfig = Pick<DataSent<ServiceConfigResponse>, 'version' | 'config' | 'clientId'>
```
Comment (ktrz-sre-is) [inline packages/node/src/hub/model.ts:7]: same as above. The `clientId` is not going to be present in the ServiceConfig response

Comment (ktrz-sre-is) [reply packages/node/src/hub/model.ts:7]: I think just removing this should be enough as you didn't actually use it anywhere. We generate the client id in `initOrThrow` function and pass it down wherever it's needed

File: packages/node/src/service/fetchServiceConfig.test.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ vi.mock('../global/serviceConfig', () => {
 })
 
 const mockServiceConfig: ServiceConfig = {
+  clientId: 'client-id',
```
Comment (ktrz-sre-is) [inline packages/node/src/service/fetchServiceConfig.test.ts:18]: this seems unnecessary as hub doesn't return client id back

File: packages/node/src/hub/api/serviceConfigResponse.ts (hunk 2 above)
Comment (ktrz-sre-is) [reply packages/node/src/hub/api/serviceConfigResponse.ts:51]: Hub schema for reference https://github.com/StanzaSystems/apis/blob/main/openapiv2/hub.swagger.json#L801-L813

File: packages/node/src/service/startPollingConfigService.ts [hunk 5]
```diff
@@ -2,12 +2,19 @@ import { fetchServiceConfig } from './fetchServiceConfig'
 import { type ServiceConfig } from '../hub/model'
 import { startPolling } from '../utils/startPolling'
 import { logger } from '../global/logger'
+import { type FetchServiceConfigOptions } from '../hub/hubService'
 
-export const startPollingServiceConfig = () => {
+export const startPollingServiceConfig = (clientId?: string) => {
```
Comment (ktrz-sre-is) [inline packages/node/src/service/startPollingConfigService.ts:7]: I think we can safely make in non optional as we always generate a client id

**Review by ktrz-sre-is — COMMENTED** [review]
Hey Carlos
I've added a couple of additional comments. If you have any questions I'd be happy to help

**Review by ktrz-sre-is — APPROVED** [review]
LGTM 🚢

**Pull request merged**